### ISSUE TITLE
Incorrect result

### ISSUE THREAD

**Issue by tomas789** [madmann91/minbool#4]
This program

```c++
#include <iostream>
#include <vector>

#include "minbool.h"

int main(int /*argc*/, char** /*argv*/) {
    using namespace std::chrono;
    using namespace minbool;

    std::vector<uint8_t> on = { 4 };
    std::vector<uint8_t> dc = { 3, 5, 6, 7, 9, 10, 11, 12, 13, 14, 15 };
    auto solution = minimize_boolean<4>(on, dc);

    std::cout << solution.size() << " terms" << std::endl;
    for (auto& term : solution) {
        std::cout << term << std::endl;
    }    
}
```

Produces following output

```
1 terms
0100
```

But the correct answer should be

```
1 terms
-1--
```

This could be verified by a [python library](https://pypi.org/project/quine-mccluskey/)

```python
from quine_mccluskey import qm
alg = qm.QuineMcCluskey()
alg.simplify(ones=[4], dc=[3, 5, 6, 7, 9, 10, 11, 12, 13, 14, 15])
```

which produces correct answer

```
{'-1--'}
```

**Comment by madmann91** [comment]
This is not incorrect:

`0100` means `(not A) * B * (not C) * (not D)`

You can check that the solution is correct for your specification:

|    | A | B | C | D | f(A, B, C, D) | (not A) * B * (not C) * (not D) |
|----|---|---|---|---|---------------|-----|
| 0  | 0 | 0 | 0 | 0 |       0       |   0  |
| 1  | 0 | 0 | 0 | 1 |       0       |   0  |
| 2  | 0 | 0 | 1 | 0 |       0       |   0  |
| 3  | 0 | 0 | 1 | 1 |       X       |   0  |
| 4  | 0 | 1 | 0 | 0 |       1       |   1  |
| 5  | 0 | 1 | 0 | 1 |       X       |   0  |
| 6  | 0 | 1 | 1 | 0 |       X       |   0  |
| 7  | 0 | 1 | 1 | 1 |       X       |   0  |
| 8  | 1 | 0 | 0 | 0 |       0       |   0  |
| 9  | 1 | 0 | 0 | 1 |       X       |   0  |
| 10 | 1 | 0 | 1 | 0 |       X       |   0  |
| 11 | 1 | 0 | 1 | 1 |       X       |   0  |
| 12 | 1 | 1 | 0 | 0 |       X       |   0  |
| 13 | 1 | 1 | 0 | 1 |       X       |   0  |
| 14 | 1 | 1 | 1 | 0 |       X       |   0  |
| 15 | 1 | 1 | 1 | 1 |       X       |   0  |

This is only maybe suboptimal in terms of operations, and the algorithm in this repository switches to heuristics anyway when no implicant chart simplification is found, leading to potentially suboptimal solutions in that case (but reducing execution time). That said, I'll try to see if I can get it to generate a better solution.

**Comment by madmann91** [comment]
Fixed in commit 31821f8
